apps/explorer/src/components/Activity/CheckpointsActivityTable.tsx

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useRpcClient } from '@mysten/core';
+import { ArrowRight12 } from '@mysten/icons';
+import { useQuery } from '@tanstack/react-query';
+import { useMemo, useState } from 'react';
+
+import { genTableDataFromCheckpointsData } from './utils';
+
+import { useGetCheckpoints } from '~/hooks/useGetCheckpoints';
+import { Link } from '~/ui/Link';
+import { Pagination } from '~/ui/Pagination';
+import { PlaceholderTable } from '~/ui/PlaceholderTable';
+import { TableCard } from '~/ui/TableCard';
+import { Text } from '~/ui/Text';
+import { numberSuffix } from '~/utils/numberUtil';
+
+const DEFAULT_CHECKPOINTS_LIMIT = 20;
+
+interface Props {
+    disablePagination?: boolean;
+    refetchInterval?: number;
+    initialLimit?: number;
+    initialCursor?: string;
+    maxCursor?: string;
+}
+
+export function CheckpointsActivityTable({
```

Comment:
Given this is used on epochs page, this naming and component location is kind of misleading. I'd maybe just put this in `components/checkpoints`.